src/circular_buffer.jl

```diff
@@ -14,29 +14,29 @@ mutable struct CircularBuffer{T} <: AbstractVector{T}
     length::Int
     buffer::Vector{T}
 
-    function CircularBuffer{T}(f,len,buf) where {T}
+    function CircularBuffer{T}(f::Int, len::Int, buf) where {T}
         f <= length(buf) || throw(ArgumentError("Value of 'first' must be inbounds of buffer"))
         len <= length(buf) || throw(ArgumentError("Value of 'length' must be <= length of buffer"))
         return new{T}(length(buf), f, len, buf)
     end
 
     # Convert any `Integer` to whatever `Int` is on the relevant machine
-    CircularBuffer{T}(f::Integer, len::Integer, buf::Integer) where {T} = CircularBuffer{T}(Int(f), Int(len), Int(buf))
```

Comment:
This constructor could be removed entirely, as it was just never called in the previous version. However it seems that https://github.com/JuliaCollections/DataStructures.jl/pull/893 decided to keep it